### ISSUE TITLE
feat: Win probability curve generation (LE-07)

### ISSUE THREAD

**Issue by CarlosDanielDev** [CarlosDanielDev/lucid-engine#15]
## Summary

Generate win/draw/loss probabilities for each position in a game, enabling a win probability chart in the consumer app.

## Requirements

```swift
public struct WinProbability: Sendable {
    public let white: Double  // 0.0 - 1.0
    public let draw: Double   // 0.0 - 1.0
    public let black: Double  // 0.0 - 1.0
}
```

### On LucidEngine actor:
```swift
public func winProbability(score: Score) -> WinProbability
```

### On GameAnalysis:
```swift
public let winProbabilities: [WinProbability]  // one per position
```

### Algorithm

Convert centipawn score to win probability using Stockfish's WDL (Win/Draw/Loss) model:
- Based on the logistic function calibrated from millions of Stockfish self-play games
- Accounts for material count (fewer pieces = higher draw probability)
- Mate scores = 100% win for the mating side

### Formula (simplified Stockfish WDL):
```
win  = 1 / (1 + exp(-0.00368208 * centipawns))
draw = estimated from piece count and score proximity to 0
loss = 1 - win - draw
```

## Acceptance Criteria

- [ ] Equal position (~0cp) returns ~50/25/25 (win/draw/loss)
- [ ] +500cp returns >90% win probability for the advantaged side
- [ ] Mate score returns 100% win, 0% draw, 0% loss
- [ ] Probabilities always sum to 1.0 (within floating point tolerance)
- [ ] GameAnalysis includes win probability for each position
- [ ] TDD: tests with known positions and expected probability ranges

## Dependencies

- Blocked by: LE-04 (game analysis pipeline)

## References

- [PRD](docs/PRD.md) — Feature LE-07
- Stockfish WDL model: lichess.org/@/lichess/blog/how-lichess-computes-win-draw-loss-probabilities
